core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -11507,6 +11510,230 @@ class KafkaApisTest extends Logging {
     assertEquals(Errors.NONE.code(), response.data.groups.get(1).errorCode())
   }
 
+  @Test
+  def testReadShareGroupStateSuccess(): Unit = {
+    val topicId = Uuid.randomUuid();
+    val readRequestData = new ReadShareGroupStateRequestData()
+      .setGroupId("group1")
+      .setTopics(List(
+        new ReadShareGroupStateRequestData.ReadStateData()
+          .setTopicId(topicId)
+          .setPartitions(List(
+            new ReadShareGroupStateRequestData.PartitionData()
+              .setPartition(1)
+              .setLeaderEpoch(1)
+          ).asJava)
+      ).asJava)
+
+    val readStateResultData: util.List[ReadShareGroupStateResponseData.ReadStateResult] = List(
+      new ReadShareGroupStateResponseData.ReadStateResult()
+        .setTopicId(topicId)
+        .setPartitions(List(
+          new ReadShareGroupStateResponseData.PartitionResult()
+            .setPartition(1)
+            .setErrorCode(Errors.NONE.code())
+            .setErrorMessage(null)
+            .setStateEpoch(1)
+            .setStartOffset(10)
+            .setStateBatches(List(
+              new ReadShareGroupStateResponseData.StateBatch()
+                .setFirstOffset(11)
+                .setLastOffset(15)
+                .setDeliveryState(0)
+                .setDeliveryCount(1)
+            ).asJava)
+        ).asJava)
+    ).asJava
+
+    val config = Map(
+      GroupCoordinatorConfig.NEW_GROUP_COORDINATOR_ENABLE_CONFIG -> "true",
```

Comment:
I believe it is the case that the new group coordinator is enabled by default in trunk now. I expect this test no longer needs to enable it.